webrtc/src/peer_connection/peer_connection_internal.rs

```diff
@@ -425,33 +421,20 @@ impl PeerConnectionInternal {
     pub(super) async fn add_transceiver_from_kind(
         &self,
         kind: RTPCodecType,
-        init: &[RTCRtpTransceiverInit],
+        init: Option<RTCRtpTransceiverInit>,
     ) -> Result<Arc<RTCRtpTransceiver>> {
         if self.is_closed.load(Ordering::SeqCst) {
             return Err(Error::ErrConnectionClosed);
         }
 
-        let direction = match init.len() {
-            0 => RTCRtpTransceiverDirection::Sendrecv,
-            1 => init[0].direction,
-            _ => return Err(Error::ErrPeerConnAddTransceiverFromKindOnlyAcceptsOne),
-        };
+        let direction = init
+            .map(|value| value.direction)
+            .unwrap_or(RTCRtpTransceiverDirection::Sendrecv);
 
         let t = match direction {
-            RTCRtpTransceiverDirection::Sendonly | RTCRtpTransceiverDirection::Sendrecv => {
-                let codecs = self.media_engine.get_codecs_by_kind(kind).await;
-                if codecs.is_empty() {
-                    return Err(Error::ErrNoCodecsAvailable);
-                }
-                let track = Arc::new(TrackLocalStaticSample::new(
-                    codecs[0].capability.clone(),
-                    math_rand_alpha(16),
-                    math_rand_alpha(16),
-                ));
-
-                self.new_transceiver_from_track(direction, track).await?
-            }
-            RTCRtpTransceiverDirection::Recvonly => {
+            RTCRtpTransceiverDirection::Sendonly
```

Comment:
The spec doesn't mention any special handling for direction https://www.w3.org/TR/webrtc/#dom-rtcpeerconnection-addtransceiver

Even for inactive the steps are the same

Reply:
@rogurotus you resolved this but didn't change the code, what's your thinking here?

Reply:
what you mean hasn't changed.

I made it possible to create an inactive transceiver.

I'm just throwing an error

```
        if direction == RTCRtpTransceiverDirection::Unspecified {
            return Err(Error::ErrPeerConnAddTransceiverFromTrackSupport);
        }
```

Maybe I somehow formatted it wrong, but everything is in commit `fix`

Reply:
Hmm, I must've been misreading then. I see it in the `fix` commit.